semgrep-core/src/fixing/tests/README.txt

```diff
@@ -0,0 +1,2 @@
+This is a separate directory to avoid a circular dependency with semgrep_engine,
+since we need to run the engine as part of the unit tests
```

Comment:
An alternative strategy to avoid the circular deps is to make Unit_autofix_printer takes an additional function called semgrep_engine and pass it in Test.ml which has access to all the modules.
I do that in matching/Unit_matcher.ml where the parser is passed as a string, to avoid having matching/ depending on parsing/